pubsub/server.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package pubsub
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/gorilla/websocket"
+
+	"go.uber.org/zap"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+)
+
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  readBufferSize,
+	WriteBufferSize: writeBufferSize,
+	CheckOrigin: func(*http.Request) bool {
+		return true
+	},
+}
+
+// Server maintains the set of active clients and sends messages to the clients.
+type Server struct {
```

Comment:
I recommend adding a small comment about how you would dial this server. I think you just use an included `gorilla` client?